net-vpn/strongswan/strongswan-6.0.3.ebuild

```diff
@@ -0,0 +1,253 @@
+# Copyright 1999-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="8"
+inherit systemd
+
+DESCRIPTION="IPsec-based VPN solution, supporting IKEv1/IKEv2 and MOBIKE"
+HOMEPAGE="https://www.strongswan.org/"
+SRC_URI="https://download.strongswan.org/${P}.tar.bz2"
+
+LICENSE="GPL-2 RSA DES"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~riscv ~x86"
+IUSE="+caps curl +constraints debug dhcp eap farp gcrypt +gmp ldap mysql networkmanager +non-root +openssl selinux sqlite systemd pam pkcs11"
+
+STRONGSWAN_PLUGINS_STD="aes cmac curve25519 des dnskey drbg eap-radius fips-prf gcm hmac led lookip md5 nonce pem pgp
+pkcs1 pkcs7 pkcs8 pkcs12 pubkey random rc2 revocation sha1 sha2 sshkey systime-fix stroke unity vici x509 xcbc"
+STRONGSWAN_PLUGINS_OPT_DISABLE="kdf"
+STRONGSWAN_PLUGINS_OPT="acert af-alg agent addrblock aesni botan blowfish bypass-lan
+ccm chapoly connmark ctr error-notify forecast files gcm ha ipseckey md4 mgf1
+openxpki padlock rdrand save-keys sha3 soup test-vectors unbound whitelist xauth-noauth"
+
+for mod in $STRONGSWAN_PLUGINS_STD; do
+	IUSE="${IUSE} +strongswan_plugins_${mod}"
+done
+
+for mod in $STRONGSWAN_PLUGINS_OPT_DISABLE; do
+	IUSE="${IUSE} strongswan_plugins_${mod}"
+done
+
+for mod in $STRONGSWAN_PLUGINS_OPT; do
+	IUSE="${IUSE} strongswan_plugins_${mod}"
+done
+
+COMMON_DEPEND="
+	non-root? (
+		acct-user/ipsec
+		acct-group/ipsec
+	)
+	dev-libs/glib:2
+	gmp? ( >=dev-libs/gmp-4.1.5:= )
+	gcrypt? (
+		dev-libs/libgcrypt:=
+		dev-libs/libgpg-error
+	)
+	caps? ( sys-libs/libcap )
+	curl? ( net-misc/curl )
+	ldap? ( net-nds/openldap:= )
+	openssl? ( >=dev-libs/openssl-0.9.8:=[-bindist(-)] )
+	mysql? ( dev-db/mysql-connector-c:= )
+	sqlite? ( >=dev-db/sqlite-3.3.1:3 )
+	systemd? ( sys-apps/systemd )
+	networkmanager? ( net-misc/networkmanager )
+	pam? ( sys-libs/pam )
+	strongswan_plugins_botan? ( dev-libs/botan:= )
```

Comment:
This needs to be `:3=`. If you have both slots installed, the dependency isn't necessarily bound to the one the binary is using. I'll fix it on merge.

If both `botan:2` and `botan:3` are installed, `dev-libs/botan:=` may be materialised as `dev-libs/botan:2/19` in the VDB (`/var/db/pkg`), yet its NEEDED may well be `libbotan.so.3`.

Reply:
Thank's. I'll investigate myself that more thoroughly because it is not completely clear for me why _here_ := just can't work.

Reply:
You as well! It's always a pleasure to review your work and to see the care you put into it.

Reply:
This particular case is slightly academic but I think I can see how it can go wrong if both Botan 2 and Botan 3 are queued (or Botan 2 is already installed), strongswan is built against Botan 2 (in terms of what Portage assumes) but because Botan 3 is available on the system & the build system prefers it, it will use that, and so the contents of `RDEPEND` in the vdb don't match what actually was linked against. I haven't tried to reproduce it though.

(The other thing is just that `x:=` doesn't mean, I think, that it has to bind to the newest version installed.)